firefox-ios/Client/Frontend/Browser/Tabs/Views/TabDisplayDiffableDataSource.swift

```diff
@@ -21,14 +21,21 @@ final class TabDisplayDiffableDataSource: UICollectionViewDiffableDataSource<Tab
 
         snapshot.appendSections([.inactiveTabs, .tabs])
 
-        // reloading .inactiveTabs is necessary to animate the caret moving when we show or hide inactive tabs
-        snapshot.reloadSections([.inactiveTabs])
-
         if state.isInactiveTabsExpanded {
             let inactiveTabs = state.inactiveTabs.map { TabDisplayViewItem.inactiveTab($0) }
             snapshot.appendItems(inactiveTabs, toSection: .inactiveTabs)
         }
 
+        // reloading .inactiveTabs is necessary to animate the caret moving when we show or hide inactive tabs
+        if #available(iOS 16, *) {
+            snapshot.reloadSections([.inactiveTabs])
+        } else {
+            let inactiveTabItems = snapshot.itemIdentifiers(inSection: .inactiveTabs)
+            if !inactiveTabItems.isEmpty {
+                snapshot.reloadSections([.inactiveTabs])
+            }
+        }
```

Comment:
If this code is only needed when users are either showing or hiding inactive tabs, what do you think about adding this in its own method and calling it when needed?

I prefer the iOS 15 version in general since it makes it explicit that we only reload sections if its available, but its nice that they updated the `reloadSections` method to handle this now. Is there documentation that supports the difference in iOS 16 / 15? Do we want to add it here if so?

Also would love to see tests as well!

Reply:
lol so I completely changed how I'm doing this - sorry! This felt too janky to me 😆 Feel free to re-review